numpyro/infer/hmc.py

```diff
@@ -485,13 +521,31 @@ def init(self, rng_key, num_warmup, init_params=None, model_args=(), model_kwarg
             raise ValueError('Valid value of `init_params` must be provided with'
                              ' `potential_fn`.')
 
+        # change dense_mass to a structural form
+        dense_mass = self._dense_mass
+        if self._model is not None:
+            z = init_params[0] if isinstance(init_params, ParamInfo) else init_params
+            if isinstance(dense_mass, bool):
+                # XXX: by default, the order variables are sorted by their names,
+                # this is to be compatible with older numpyro versions
+                # and to match autoguide scale parameter and jax flatten utils
+                dense_mass = [tuple(sorted(z))] if dense_mass else []
+            else:
+                # assert that site_names in dense_mass and user-defined
+                # inverse_mass_matrix matched
+                assert isinstance(dense_mass, list)
+                if isinstance(self._inverse_mass_matrix, dict):
+                    for site_names in dense_mass:
+                        assert site_names in self._inverse_mass_matrix
```

Comment:
+ error message?

Reply:
We will check for consistency in `_initialize_mass_matrix` method and will raise the error there. So I would like to remove this branch.